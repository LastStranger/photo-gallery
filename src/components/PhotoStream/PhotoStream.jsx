// @flow

import { h, Component } from 'preact';
import PhotoStreamPhoto from '../PhotoStreamPhoto';
import Header from '../Header';
import './PhotoStream.css';

const generatePlaceHolders = (max: number): Array<Photo> => {
  const answer = [];
  for (let i = 0; i < max; i += 1) {
    answer.push({
      id: i.toString(),
      small: '/assets/placeholder_small.jpg',
      medium: '/assets/placeholder_medium.jpg',
      large: '/assets/placeholder_large.jpg',
      title: 'placeholder'
    });
  }
  return answer;
};

class PhotoStream extends Component {
  state = {
    photos: generatePlaceHolders(20)
  };
  componentDidMount() {
    fetch('/flickr/photos/dogs/1').then(response => response.json()).then((flickrRes: FlickrPhotosResponse) => {
      this.setState({ photos: flickrRes.photos });
      this.props.setLoading(false);
    });
  }
  componentWillUnmount() {
    this.startLoading();
  }
  startLoading = () => {
    this.props.setLoading(true);
  };
  props: {
    setLoading: Function
  };
  render() {
    return (
      <div className="photo-stream">
        <Header showSpacer title="Image gallery" />
        <div className="photo-stream-photos">
          {this.state.photos.map(photo =>
            <PhotoStreamPhoto
              startLoading={this.startLoading}
              key={photo.id}
              id={photo.id}
              url={photo.small}
              alt={photo.title}
              show
            />
          )}
        </div>
      </div>
    );
  }
}

export default PhotoStream;
