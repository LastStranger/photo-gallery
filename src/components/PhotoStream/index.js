// @flow
import PhotoStream from './PhotoStream';

export default PhotoStream;
