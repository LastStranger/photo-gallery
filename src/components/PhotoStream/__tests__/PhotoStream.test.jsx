// @flow

import { h } from 'preact';
import render from 'preact-render-to-string';
import PhotoStream from '../';
import { photos } from '../../../testHelpers';

test('renders', () => {
  const string = render(<PhotoStream setLoading={() => {}} />);
  expect(string).toMatchSnapshot();
});

test('makes the correct ajax call', () => {
  global.fetch = jest.fn().mockImplementation(() => Promise.resolve({ json: () => photos }));
  const instance = new PhotoStream();
  instance.props = { setLoading: () => {} };
  instance.componentDidMount();
  expect(global.fetch).toBeCalledWith('/flickr/photos/dogs/1');
});
