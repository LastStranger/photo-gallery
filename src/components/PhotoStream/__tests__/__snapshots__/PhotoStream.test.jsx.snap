// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div class="photo-stream">
  <div class="header-container spacer">
    <header class="header-bar">
      <h1 class="header-title">Image gallery</h1>
    </header>
  </div>
  <div class="photo-stream-photos">
    <a class="stream-photo" href="/0">
      <div aria-label="placeholder" role="img" class="stream-photo-img" style="background-image: url(/assets/placeholder_small.jpg);"></div>
    </a>
    <a class="stream-photo" href="/1">
      <div aria-label="placeholder" role="img" class="stream-photo-img" style="background-image: url(/assets/placeholder_small.jpg);"></div>
    </a>
    <a class="stream-photo" href="/2">
      <div aria-label="placeholder" role="img" class="stream-photo-img" style="background-image: url(/assets/placeholder_small.jpg);"></div>
    </a>
    <a class="stream-photo" href="/3">
      <div aria-label="placeholder" role="img" class="stream-photo-img" style="background-image: url(/assets/placeholder_small.jpg);"></div>
    </a>
    <a class="stream-photo" href="/4">
      <div aria-label="placeholder" role="img" class="stream-photo-img" style="background-image: url(/assets/placeholder_small.jpg);"></div>
    </a>
    <a class="stream-photo" href="/5">
      <div aria-label="placeholder" role="img" class="stream-photo-img" style="background-image: url(/assets/placeholder_small.jpg);"></div>
    </a>
    <a class="stream-photo" href="/6">
      <div aria-label="placeholder" role="img" class="stream-photo-img" style="background-image: url(/assets/placeholder_small.jpg);"></div>
    </a>
    <a class="stream-photo" href="/7">
      <div aria-label="placeholder" role="img" class="stream-photo-img" style="background-image: url(/assets/placeholder_small.jpg);"></div>
    </a>
    <a class="stream-photo" href="/8">
      <div aria-label="placeholder" role="img" class="stream-photo-img" style="background-image: url(/assets/placeholder_small.jpg);"></div>
    </a>
    <a class="stream-photo" href="/9">
      <div aria-label="placeholder" role="img" class="stream-photo-img" style="background-image: url(/assets/placeholder_small.jpg);"></div>
    </a>
    <a class="stream-photo" href="/10">
      <div aria-label="placeholder" role="img" class="stream-photo-img" style="background-image: url(/assets/placeholder_small.jpg);"></div>
    </a>
    <a class="stream-photo" href="/11">
      <div aria-label="placeholder" role="img" class="stream-photo-img" style="background-image: url(/assets/placeholder_small.jpg);"></div>
    </a>
    <a class="stream-photo" href="/12">
      <div aria-label="placeholder" role="img" class="stream-photo-img" style="background-image: url(/assets/placeholder_small.jpg);"></div>
    </a>
    <a class="stream-photo" href="/13">
      <div aria-label="placeholder" role="img" class="stream-photo-img" style="background-image: url(/assets/placeholder_small.jpg);"></div>
    </a>
    <a class="stream-photo" href="/14">
      <div aria-label="placeholder" role="img" class="stream-photo-img" style="background-image: url(/assets/placeholder_small.jpg);"></div>
    </a>
    <a class="stream-photo" href="/15">
      <div aria-label="placeholder" role="img" class="stream-photo-img" style="background-image: url(/assets/placeholder_small.jpg);"></div>
    </a>
    <a class="stream-photo" href="/16">
      <div aria-label="placeholder" role="img" class="stream-photo-img" style="background-image: url(/assets/placeholder_small.jpg);"></div>
    </a>
    <a class="stream-photo" href="/17">
      <div aria-label="placeholder" role="img" class="stream-photo-img" style="background-image: url(/assets/placeholder_small.jpg);"></div>
    </a>
    <a class="stream-photo" href="/18">
      <div aria-label="placeholder" role="img" class="stream-photo-img" style="background-image: url(/assets/placeholder_small.jpg);"></div>
    </a>
    <a class="stream-photo" href="/19">
      <div aria-label="placeholder" role="img" class="stream-photo-img" style="background-image: url(/assets/placeholder_small.jpg);"></div>
    </a>
  </div>
</div>
`;
